### DISCUSSION TITLE
Add ability to hide/arrange default fields on the Directus Files edit view

### DISCUSSION THREAD

**Discussion by jeremyzilar** [directus/directus#23076] [Feature Requests]
### Summary

As an Editor in Directus, I need easier access to the additional fields we have configured in the `directus_files` schema. 
Our small team of editors process and upload 200-300 image each month, they need go into the edit view multiple times to make edits to captions.

At the moment, the fields we added in the schema are at the bottom of the "files" edit screen and there is no way to move them up higher, or hide the default Directus fields we are not using. It would be nice to be able to hide some of the default fields and/or be able to sort/arrange the our fields with the default fields.

https://github.com/user-attachments/assets/d8c2abc0-6fb3-4b24-8099-9b05a7f0adc9

### Motivation

This would make it easier for people to use and customize the `directus_files` content type in their workflows. As a site designer, there are a number of fields that our publication will need for each image.
- caption -- this would need to be a WYSIWYG field that supports bold, italic, links. Currently the "description" does not support these.
- alt text -- we are going to want to have a field for writing out descriptive alt text for our images. In arts organizations, it is increasingly becoming important to write the ALT text differently than the caption, which can include less information because the sighted folks have the benefit of seeing the image.
- license information -- some images are only licensed to run on the web, in connection with this one article. Not in marketing materials or print.
- Photo credit -- it would be nice to credit the photographer in a data-first way so that we can easily reuturn all of the images shot by a single person.

Unfortunately, any additional field we add is really inaccessible to an editor that is moving pretty fast to produce content for the website.

### Detailed Design

This would be a change to the existing implementation of the `directus_files` interface to allow for the default fields to either be hidden or sorted.

### Must Have:
- allow for the default `directus_files` fields to either be hidden or be sortable.

### Drawbacks

TO be honest, I am not sure what the implications are of making default fields sortable or hideable.

### Alternatives

I thought about just creating another content type for our images, but that created a lot of additional complexity in the content type, which would still have to rely on `directus_files`. And it made it harder to create upload multiple files into a single field and create relations back to the images/

### Adoption Strategy

I would not think that this is a breaking change. I feel like folks would suddenly realize that they can edt the schema in `directus_files`.

**Comment by jeremyzilar** [comment]
I also tried hiding these fields through CSS, but doesn't seem to be unique enough classnames or IDs to hang off of in that files edit panel.

**Comment by saneruby** [comment]
Hi! I also have this problem. There are a lot of fields in the directus_users collection that need to be filled out. In addition to name and last name, there are extra fields. People have to look through a lot to find the right fields. For example, there's a role called client manager. His job is to give clients access. I can't hide fields like API key, location, theme and description.

**Reply by gloribus** [reply]
https://github.com/directus/directus/discussions/19529

**Comment by gloribus** [comment]
Two years ago this was already brought up - and since then, we even got the "Topics" field that can't be hidden and takes up an absurd amount of space in the user card.  
You can't move it to a right-hand drawer, you can't tuck it into a tab, you can't collapse it, or relocate it. I get that this might be necessary at the database level - but why can't we just visually manage it? Like simply hiding or repositioning fields?

It doesn’t affect functionality at all — I can already restrict these fields from being filled out via the role system. So this feels like an artificial limitation, not a technical one.

In many setups, users are the ones filling out their own data. And when they're faced with a never-ending list of fields — some of which (like "Topics") eat up the entire screen — it’s overwhelming. It scares people off.

https://github.com/directus/directus/discussions/19529

**Comment by rafarc21** [comment]
Agreed. This is also a pain for us.

**Comment by Mehdi-YC** [comment]
Same here, i needed to change it for a specific role, only , but i can't hide them like in other ones

**Comment by ThePr0grammer** [comment]
Same here. It's understanable that you can't delete fields that considered part of a system, to not break anything. But why i can't just hide it from view? Very need feature.

**Comment by fergusmurray** [comment]
Plus one for this. Adding Users is far more painful and complex for end users than it should be.
Accordion and moving field positions would be much appreciated.

**Comment by domdus** [comment]
Hi all, 

I recently released this extension to address the cases you mentioned above:
https://github.com/domdus/directus-extension-system-fields-manager
(It is in Marketplace too)

You can hide & reorder fields the way you need it - per `role` and/or `policy`.

I tested quite a lot, but in case of bugs let me know. 😅
